src/librustc/middle/intrinsicck.rs

```diff
@@ -67,20 +67,6 @@ impl<'a, 'gcx, 'tcx> ExprVisitor<'a, 'gcx, 'tcx> {
             if sk_from.same_size(sk_to) {
                 return;
             }
-
-            match (&from.sty, sk_to) {
-                (&ty::TyFnDef(..), SizeSkeleton::Known(size_to))
-                        if size_to == Pointer.size(&self.infcx.tcx.data_layout) => {
-                    // FIXME #19925 Remove this warning after a release cycle.
-                    let msg = format!("`{}` is now zero-sized and has to be cast \
-                                       to a pointer before transmuting to `{}`",
-                                      from, to);
-                    self.infcx.tcx.sess.add_lint(
-                        ::lint::builtin::TRANSMUTE_FROM_FN_ITEM_TYPES, id, span, msg);
-                    return;
-                }
-                _ => {}
```

Comment:
TBH I still find it hard to believe that *anyone* needs to ever transmute function pointers.
We maybe need to allow `unsafe extern "C"` -variadic function types that you can coerce to from your own function types - since that is the one case where you can't use a polymorphic wrapper.